Singularity/Singularity/Resources/Resource.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Singularity.Utils;
+
+namespace Singularity.Resources
+{
+    /// <inheritdoc cref="IResource"/>
+    /// <summary>
+    /// Represents a resource in the game. Written in such a fashion that it can represent any resource there is and will be.
+    /// </summary>
+    internal class Resource : IResource
+    {
+        /// <summary>
+        /// The velocity of this resource object.
+        /// </summary>
+        private const float Velocity = 2;
+
+        /// <summary>
+        /// The width for the texture to be rendered.
+        /// </summary>
+        private const int Width = 10;
+
+        /// <summary>
+        /// The height for the texture to be rendered.
+        /// </summary>
+        private const int Height = 10;
+
+        /// <summary>
+        /// The current position of this resource on the map.
+        /// </summary>
+        private Vector2 mPosition;
+
+        /// <summary>
+        /// The direction the resource is moving.
+        /// </summary>
+        private Vector2 mDirection;
+
+        /// <summary>
+        /// The texture of this resource.
+        /// </summary>
+        private Texture2D mTexture;
+
+        /*
+           TODO: im not quite sure whether the ID is "unique" such that every instance of this class
+           TODO: should hold a different id, or if every different resouce should hold a different id,
+           TODO: thus i've not written any id assignment yet.
+        */
+        public int Id { get; }
+
+        public EResourceType Type { get; }
```

Comment:
Is this actually compiling? how does it now what value to return etc? - but could absolutely be a mistake on my side here

Reply:
Yeah it sure does, it basically defines an invisible member variable and uses it for the properties inherited by the interface (atleast thats what I read).